### PR TITLE
Updating Service Catalog section to explain Service constraints.

### PR DESCRIPTION
**Pull request by dperaza4dustbit**
Adding more details to the service catalog section explaining how each service catalog for each cluster environment gets update when there is a change to a registered service.

### CODE REVIEW

File: docs/entities/servicecatalog.md [hunk 1]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
 
 ### Update
 
-When a registered service is being updated the Service Catalog will get updated too.
-The updated Sevice Catalog will be made available to all the application namespaces of matching Cluster Environment by Primaza.
+When a registered service is being updated, the appropriate service catalog will get updated too.
+The updated service catalog will be made available to all the application namespaces of matching cluster Eenvironment by Primaza.
```
Comment (filariow) [inline docs/entities/servicecatalog.md:34]:
```suggestion
The updated Service Catalog will be made available to all the application namespaces of matching Cluster Environment by Primaza.
```

File: docs/entities/servicecatalog.md [hunk 2]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
...
+The constraint section of a registered service defines a list of environments.
```
Comment (filariow) [inline docs/entities/servicecatalog.md:35]:
```suggestion
The `constraints` section of a registered service defines a list of environments.
```

File: docs/entities/servicecatalog.md [hunk 3]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
...
+The environment list can contain explicit environments that are to include the service or it can also contain a list of environments that are to exclude the service.
+The way to depict exclusion is by adding the ! prefix to the environment, i.e !prod.
```
Comment (filariow) [inline docs/entities/servicecatalog.md:37]: For readability I'd propose to use code blocks for environment constraints. Applies also to next sentences.

```suggestion
The way to depict exclusion is by adding the `!` prefix to the environment, i.e `!prod`.
```

File: docs/entities/servicecatalog.md [hunk 4]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
...
+Exclusion has precedence over inclusion for example, if the list contains !prod but also includes dev, then dev is considered to be in the !prod set of environments and therefore redundant.
```
Comment (Kartikey-star) [inline docs/entities/servicecatalog.md:38]: Line 38 should probably come with a `Please Note`

**Review by Kartikey-star — DISMISSED** [review]
Looks good to me. Just an optional nit.

**Review by filariow — CHANGES_REQUESTED** [review]
a typo and a some proposals for readability

**Review by sadlerap — CHANGES_REQUESTED** [review]
Some more nits

File: docs/entities/servicecatalog.md (hunk 4 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:38]: OK

File: docs/entities/servicecatalog.md (hunk 1 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:34]: OK

File: docs/entities/servicecatalog.md (hunk 2 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:35]: OK

File: docs/entities/servicecatalog.md (hunk 3 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:37]: OK

File: docs/entities/servicecatalog.md [hunk 5]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
...
+If there is a third environment stage then !prod would include both stage and dev even if they are not defined in the list explicitly.
```
[the remark this answers is not in the document]
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:39]: OK

File: docs/entities/servicecatalog.md [hunk 6]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
 
 ### Update
 
-When a registered service is being updated the Service Catalog will get updated too.
-The updated Sevice Catalog will be made available to all the application namespaces of matching Cluster Environment by Primaza.
+When a registered service is being updated, the appropriate service catalog will get updated too.
```
[the remark this answers is not in the document]
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:33]: OK

File: docs/entities/servicecatalog.md [hunk 7]
```diff
@@ -30,5 +30,12 @@ The Catalog will also be removed from the matching application namepsaces.
...
+The updated service catalog will be made available to all the application namespaces of matching cluster Eenvironment by Primaza.
+The constraint section of a registered service defines a list of environments.
+The environment list can contain explicit environments that are to include the service or it can also contain a list of environments that are to exclude the service.
```
[the remark this answers is not in the document]
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:36]: OK

File: docs/entities/servicecatalog.md (hunk 1 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md:34]: made the change with the typo, but should be consistent with Upper Case for Resource name vs lower case. For example Service Catalog vs service catalog or Registered Service vs registered service. I think we should stick with lower case unless you are referring to the K8 name which will be ServiceCatalog and RegisteredService.

**Review by Kartikey-star — APPROVED** [review]

File: docs/entities/servicecatalog.md [hunk 8]
```diff
@@ -30,5 +30,13 @@ The Catalog will also be removed from the matching application namepsaces.
...
 ### Update
 
-When a registered service is being updated the Service Catalog will get updated too.
-The updated Sevice Catalog will be made available to all the application namespaces of matching Cluster Environment by Primaza.
+When a registered service is updated, the appropriate service catalog will get updated as well.
+The updated service catalog will be made available to all the application namespaces of matching cluster environment by Primaza.
+The `constraints` section of a registered service defines a list of environments.
+The environment list can contain explicit environments that are to include the service, or it can also contain a list of environments that are to exclude the service.
+The way to depict exclusion is by adding the `!` prefix to the environment, i.e `!prod`.
+Please note that exclusions has precedence over inclusions.
```
Comment (filariow) [inline docs/entities/servicecatalog.md:38]:
```suggestion
Please note that exclusions have precedence over inclusions.
```

**Review by filariow — CHANGES_REQUESTED** [review]
just one small nit

File: docs/entities/servicecatalog.md (hunk 8 above)
Comment (dperaza4dustbit) [reply docs/entities/servicecatalog.md]: done

**Review by filariow — APPROVED** [review]
/lgtm

**Review by sadlerap — APPROVED** [review]

**Pull request merged**